sql/privileges.go

```diff
@@ -56,6 +62,10 @@ type PrivilegedOperationChecker interface {
 	// privileged operation(s). This takes into account the active roles, which are set in the context, therefore both
 	// the user and the active roles are pulled from the context.
 	UserHasPrivileges(ctx *Context, operations ...PrivilegedOperation) bool
+	// UserHasExplicitRoutinePrivileges fetches the User from the context, and specifically evaluates, the permission check
+	// assuming the operation is for a stored procedure or function. This allows us to have more fine grain control over
+	// permissions for stored procedures (many of which are critical to Dolt).
+	UserHasExplicitRoutinePrivileges(ctx *Context, operations ...PrivilegedOperation) bool
```

Comment:
I think contrasting naming might make the difference between these two more obvious. `Has[User,Admin]RoutinePrivileges`, `Has[User,Admin]RoutineAccess`, `[User,Admin]RoutineAccess`, ....

Reply:
Can you say more here? Are you suggesting splitting out the super user check? I understand dropping the "User" from the names, but I don't think that results in a contrast as you are requesting.

Reply:
Even though I have context on the difference, to me these sound like the same thing
```golang
	UserHasPrivileges(ctx *Context, operations ...PrivilegedOperation) bool
	UserHasExplicitRoutinePrivileges(ctx *Context, operations ...PrivilegedOperation) bool
```

So I think there is an opportunity for them (1) not to sound similar, (2) sound different in ways that meaningfully describe what they do differently

Reply:
This sounds like renaming "UserHasPrivleges" because UserHasExplicitRoutinePrivileges is quite specific.

Refactoring the first should be a mechanical operation done by goland which should be a seperate PR.

Reply:
I think this is fairly important, interface shape and names are usually the longest lasting changes of any PR.  I'm not sure I agree that doing a second set of GMS and Dolt PRs is the most time-efficient way to do this, but as long as we don't forget and it ends up in backlog land. If it wasn't clear, my suggestion was renaming to suggest "user" and "admin" procs that are authorized differently: `UserHasPrivileges-> HasUserRoutinePrivileges`, `UserHasExplicitRoutinePrivileges-> HasAdminRoutinePrivileges`. Not sure if that's your intent, but it's the first thing that came to mind.

Reply:
After looking through the PR, I do agree with Max here. From a naming standpoint, `UserHasPrivileges` and `UserHasExplicitRoutinePrivileges` seem quite similar. If I'm checking for routine privileges, won't `UserHasPrivileges` suffice? For another engineer to come across the code and implement these checks elsewhere, I can definitely see potential misuse in the future.

Another suggestion for the new name could be `UserHasAdminPrivilegesOnRoutines` or something.

Reply:
I'm convinced. will call it "RoutineAdminCheck" and add lots of comments on the interfaces to clarify why that exists.